web3swift/src/Client/EthereumClientProtocol.swift

```diff
@@ -0,0 +1,156 @@
+    func eth_getTransactionCount(address: EthereumAddress, block: EthereumBlock) async throws -> Int
+    func eth_getTransaction(byHash txHash: String) async throws -> EthereumTransaction
+    func eth_getTransactionReceipt(txHash: String) async throws -> EthereumTransactionReceipt
+    func eth_call(
+        _ transaction: EthereumTransaction,
+        block: EthereumBlock
+    ) async throws -> String
+    func eth_call(
+        _ transaction: EthereumTransaction,
+        resolution: CallResolution,
+        block: EthereumBlock
+    ) async throws -> String
+    func eth_getLogs(addresses: [EthereumAddress]?, topics: [String?]?, fromBlock: EthereumBlock, toBlock: EthereumBlock) async throws ->  [EthereumLog]
+    func eth_getLogs(addresses: [EthereumAddress]?, orTopics: [[String]?]?, fromBlock: EthereumBlock, toBlock: EthereumBlock) async throws ->  [EthereumLog]
+    func eth_getBlockByNumber(_ block: EthereumBlock) async throws -> EthereumBlockInfo
+
+    // Deprecated
+    func net_version(completion: @escaping((EthereumClientError?, EthereumNetwork?) -> Void))
+    func eth_gasPrice(completion: @escaping((EthereumClientError?, BigUInt?) -> Void))
+    func eth_blockNumber(completion: @escaping((EthereumClientError?, Int?) -> Void))
+    func eth_getBalance(address: EthereumAddress, block: EthereumBlock, completion: @escaping((EthereumClientError?, BigUInt?) -> Void))
+    func eth_getCode(address: EthereumAddress, block: EthereumBlock, completion: @escaping((EthereumClientError?, String?) -> Void))
+    func eth_estimateGas(_ transaction: EthereumTransaction, completion: @escaping((EthereumClientError?, BigUInt?) -> Void))
+    func eth_sendRawTransaction(_ transaction: EthereumTransaction, withAccount account: EthereumAccountProtocol, completion: @escaping((EthereumClientError?, String?) -> Void))
+    func eth_getTransactionCount(address: EthereumAddress, block: EthereumBlock, completion: @escaping((EthereumClientError?, Int?) -> Void))
+    func eth_getTransaction(byHash txHash: String, completion: @escaping((EthereumClientError?, EthereumTransaction?) -> Void))
+    func eth_getTransactionReceipt(txHash: String, completion: @escaping((EthereumClientError?, EthereumTransactionReceipt?) -> Void))
+    func eth_call(
+        _ transaction: EthereumTransaction,
+        resolution: CallResolution,
+        block: EthereumBlock,
+        completion: @escaping((EthereumClientError?, String?) -> Void)
+    )
+    func eth_getLogs(addresses: [EthereumAddress]?, topics: [String?]?, fromBlock: EthereumBlock, toBlock: EthereumBlock, completion: @escaping((EthereumClientError?, [EthereumLog]?) -> Void))
+    func eth_getLogs(addresses: [EthereumAddress]?, orTopics: [[String]?]?, fromBlock: EthereumBlock, toBlock: EthereumBlock, completion: @escaping((EthereumClientError?, [EthereumLog]?) -> Void))
+    func eth_getBlockByNumber(_ block: EthereumBlock, completion: @escaping((EthereumClientError?, EthereumBlockInfo?) -> Void))
```

Comment:
I think for this change it's ok to just remove these APIs as we have the result-based ones; to avoid duplicating code in WSSClient